### PR TITLE
Added time range and item response time component

### PR DESCRIPTION
**Pull request by VladyslavAnokhin**
## Changes

* Save response time into component
* Update HardTimeout action with items which were timeouted

[JIRA Ticket](https://jira.ouroath.com/browse/OMSDK-2213)

@VerizonAdPlatforms/mobile-sdk-developers: Please review.

### CODE REVIEW

File: PlayerCore/components/New VRM Core/VRMItemResponseTime.swift [hunk 1]
```diff
@@ -0,0 +1,78 @@
...
+
+func reduce(state: VRMItemResponseTime, action: Action) -> VRMItemResponseTime {
+    
+    func updateFinishTime(for item: VRMCore.Item,
+                          with date: Date,
+                          in container: VRMItemResponseTime.Container) -> VRMItemResponseTime.Container {
+        guard var range = container[item] else {
+            fatalError("get finish error action for item which wasn't started")
+        }
+        range.finishAt = date
```
Comment (AndriiMoskvin) [inline PlayerCore/components/New VRM Core/VRMItemResponseTime.swift]: move this one line down?

File: PlayerCore/components/New VRM Core/VRMItemResponseTime.swift [hunk 2]
```diff
@@ -0,0 +1,78 @@
...
+    case let failedParse as VRMCore.ParsingError:
+        let newContainer = updateFinishTime(for: failedParse.originalItem,
+                                            with: failedParse.finishDate,
+                                            in: state.timeRangeContainer)
+        return VRMItemResponseTime(timeRangeContainer: newContainer )
+    case let indirectionError as VRMCore.TooManyIndirections:
+        let newContainer = updateFinishTime(for: indirectionError.item,
+                                            with: indirectionError.finishDate,
+                                            in: state.timeRangeContainer)
+        return VRMItemResponseTime(timeRangeContainer: newContainer )
```
Comment (AndriiMoskvin) [inline PlayerCore/components/New VRM Core/VRMItemResponseTime.swift]: Make sense to move this inside function - no reason to duplicate it each time.

File: PlayerCore/components/New VRM Core/VRMItemResponseTime.swift [hunk 3]
```diff
@@ -0,0 +1,78 @@
...
+    func updateFinishTime(for item: VRMCore.Item,
+                          with date: Date,
+                          in container: VRMItemResponseTime.Container) -> VRMItemResponseTime.Container {
+        guard var range = container[item] else {
+            fatalError("get finish error action for item which wasn't started")
+        }
+        range.finishAt = date
+        var newTimeContainer = container
+        newTimeContainer[item] = range
+
```
Comment (AndriiMoskvin) [inline PlayerCore/components/New VRM Core/VRMItemResponseTime.swift]: Do we need this empty line before return?

File: PlayerCore/components/New VRM Core/VRMItemResponseTime.swift [hunk 4]
```diff
@@ -0,0 +1,78 @@
...
+                                            with: selectInlineVAST.date,
+                                            in: state.timeRangeContainer)
+        return VRMItemResponseTime(timeRangeContainer: newContainer )
+    case let startFetching as VRMCore.StartItemFetch:
+        return updateStartTime(for: startFetching.originalItem, with: startFetching.startDate)
+    case let startParsing as VRMCore.StartItemParsing where state.timeRangeContainer[startParsing.originalItem] == nil:
+        return updateStartTime(for: startParsing.originalItem, with: startParsing.startDate)
+    case let hardTimeout as VRMCore.HardTimeout:
+        var newTimeContainer = state.timeRangeContainer
+        hardTimeout.items.forEach { item in
```
Comment (AndriiMoskvin) [inline PlayerCore/components/New VRM Core/VRMItemResponseTime.swift]: item can be omitted

Comment (VladyslavAnokhin) [reply PlayerCore/components/New VRM Core/VRMItemResponseTime.swift]: Done

File: PlayerCore/components/New VRM Core/VRMItemResponseTime.swift (hunk 3 above)
Comment (VladyslavAnokhin) [reply PlayerCore/components/New VRM Core/VRMItemResponseTime.swift]: Removed

File: PlayerCore/components/New VRM Core/VRMItemResponseTime.swift (hunk 2 above)
Comment (VladyslavAnokhin) [reply PlayerCore/components/New VRM Core/VRMItemResponseTime.swift]: I've made this for `HardTimeout` case. There we have array of items.

Comment (AndriiMoskvin) [reply PlayerCore/components/New VRM Core/VRMItemResponseTime.swift]: Than for hard timeout write additional code - but 90% of other cases are will be covered.

File: PlayerCore/components/New VRM Core/VRMItemResponseTime.swift [hunk 5]
```diff
@@ -0,0 +1,79 @@
...
+import Foundation
+
+public struct VRMItemResponseTime {
+    typealias Container = [VRMCore.Item: TimeRange]
+    
+    static let initial = VRMItemResponseTime(timeRangeContainer: [:])
+    
+    public struct TimeRange: Hashable {
+        public let startAt: Date
+        public var finishAt: Date?
```
Comment (AndriiMoskvin) [inline PlayerCore/components/New VRM Core/VRMItemResponseTime.swift]: Does it makes sense to make it `var?`

File: PlayerCore/components/New VRM Core/VRMItemResponseTime.swift [hunk 6]
```diff
@@ -0,0 +1,79 @@
...
+    case let startFetching as VRMCore.StartItemFetch:
+        return updateStartTime(for: startFetching.originalItem,
+                               with: startFetching.startDate)
+        
+    case let startParsing as VRMCore.StartItemParsing where state.timeRangeContainer[startParsing.originalItem] == nil:
+        return updateStartTime(for: startParsing.originalItem,
+                               with: startParsing.startDate)
+        
+    case let hardTimeout as VRMCore.HardTimeout:
+        return hardTimeout.items.reduce(state) { result, item in
```
Comment (AndriiMoskvin) [inline PlayerCore/components/New VRM Core/VRMItemResponseTime.swift]: 👍

File: PlayerCore/components/New VRM Core/VRMItemResponseTime.swift (hunk 5 above)
Comment (VladyslavAnokhin) [reply PlayerCore/components/New VRM Core/VRMItemResponseTime.swift]: Yes.

**Pull request merged**